crates/next-core/src/next_config.rs

```diff
@@ -0,0 +1,216 @@
+    Cloudinary,
+    Akamai,
+    Custom,
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs)]
+pub enum ImageFormat {
+    #[serde(rename(deserialize = "image/webp"))]
+    Webp,
+    #[serde(rename(deserialize = "image/avif"))]
+    Avif,
+}
+
+#[derive(
+    Clone, Debug, Default, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs,
+)]
+#[serde(rename_all = "camelCase")]
+pub struct RemotePattern {
+    pub protocol: Option<RemotePatternProtocal>,
+    pub hostname: String,
+    pub port: Option<String>,
+    pub pathname: Option<String>,
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(rename_all = "lowercase")]
+pub enum RemotePatternProtocal {
+    Http,
+    Https,
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(rename_all = "camelCase")]
+pub struct ExperimentalConfig {
+    pub server_components_external_packages: Option<Vec<String>>,
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(rename_all = "camelCase")]
+pub struct CompilerConfig {
+    pub react_remove_properties: Option<bool>,
+    pub relay: Option<RelayConfig>,
+    pub remove_console: Option<RemoveConsoleConfig>,
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(untagged, rename_all = "camelCase")]
+pub enum ReactRemoveProperties {
+    Boolean(bool),
+    Config { properties: Option<Vec<String>> },
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(rename_all = "camelCase")]
+pub struct RelayConfig {
+    pub src: String,
+    pub artifact_directory: Option<String>,
+    pub language: Option<RelayLanguage>,
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(untagged, rename_all = "lowercase")]
+pub enum RelayLanguage {
+    TypeScript,
+    Flow,
+    JavaScript,
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, PartialEq, Eq, Serialize, Deserialize, TraceRawVcs)]
+#[serde(untagged)]
+pub enum RemoveConsoleConfig {
+    Boolean(bool),
+    Config { exclude: Option<Vec<String>> },
+}
+
+#[turbo_tasks::function]
+pub async fn load_next_config(
+    context: AssetContextVc,
+    chunking_context: ChunkingContextVc,
+    project_root: FileSystemPathVc,
+    intermediate_output_path: FileSystemPathVc,
+) -> Result<NextConfigValueVc> {
+    let mut chunks = None;
+    let next_config_mjs_path = project_root.join("next.config.mjs");
+    let next_config_js_path = project_root.join("next.config.js");
+    if let Some(config_asset) = if matches!(
+        &*next_config_mjs_path.read().await?,
+        FileContent::Content(_)
+    ) {
+        Some(SourceAssetVc::new(next_config_mjs_path))
+    } else if matches!(&*next_config_js_path.read().await?, FileContent::Content(_)) {
```

Comment:
Nit: using `path.get_type()` will allow us to see if the file exists without actually reading the file content. +1 laziness.